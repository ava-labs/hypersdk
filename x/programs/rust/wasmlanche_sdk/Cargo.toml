[package]
name = "wasmlanche_sdk"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
thiserror = "1.0.46"
sdk_macros = { version = "0.1.0", path = "../sdk_macros" }
borsh = { version = "1.2.0", features = ["derive"] }

# TODO: delete serde dependencies(simulator needs update) 
serde = { version = "1.0.185", features = ["derive"] }
serde_bare = "0.5.0"

# simulator
serde_json = { version = "1.0.107", optional = true }

[features]
default = []
simulator = ["serde_json"]

[lib]
