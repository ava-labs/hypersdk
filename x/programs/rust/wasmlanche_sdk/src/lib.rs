#![deny(clippy::pedantic)]

pub mod errors;
pub mod host;
pub mod store;
pub mod types;
pub use sdk_macros::public;
