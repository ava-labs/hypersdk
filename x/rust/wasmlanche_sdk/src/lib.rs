pub mod errors;
pub mod host;
pub mod program;
pub mod store;
pub mod types;
