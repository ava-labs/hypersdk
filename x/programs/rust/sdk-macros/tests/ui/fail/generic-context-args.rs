use wasmlanche_sdk::{borsh::BorshDeserialize, public};

#[derive(BorshDeserialize)]
#[borsh(crate = "wasmlanche_sdk::borsh")]
pub struct Context<T>(T);

#[public]
pub fn always_true(_: &mut Context<u8>) -> bool {
    true
}

fn main() {}
