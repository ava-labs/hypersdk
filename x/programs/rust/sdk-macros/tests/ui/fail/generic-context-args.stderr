error[E0308]: mismatched types
 --> tests/ui/fail/generic-context-args.rs:7:1
  |
7 | #[public]
  | ^^^^^^^^^
  | |
  | expected `&mut Context`, found `&mut Context<u8>`
  | arguments to this function are incorrect
  |
  = note: `Context<u8>` and `wasmlanche_sdk::Context` have similar names, but are actually distinct types
note: `Context<u8>` is defined in the current crate
 --> tests/ui/fail/generic-context-args.rs:5:1
  |
5 | pub struct Context<T>(T);
  | ^^^^^^^^^^^^^^^^^^^^^
note: `wasmlanche_sdk::Context` is defined in crate `wasmlanche_sdk`
 --> $WORKSPACE/x/programs/rust/wasmlanche-sdk/src/context.rs
  |
  | pub struct Context {
  | ^^^^^^^^^^^^^^^^^^
note: function defined here
 --> tests/ui/fail/generic-context-args.rs:8:8
  |
7 | #[public]
  | ---------
8 | pub fn always_true(_: &mut Context<u8>) -> bool {
  |        ^^^^^^^^^^^
  = note: this error originates in the attribute macro `public` (in Nightly builds, run with -Z macro-backtrace for more info)
